compiler/rustc_const_eval/src/interpret/place.rs

```diff
@@ -793,6 +793,16 @@ where
         }
     }
 
+    pub fn write_uninit(&mut self, dest: &PlaceTy<'tcx, M::PointerTag>) -> InterpResult<'tcx> {
+        let mplace = self.force_allocation(dest)?;
```

Comment:
`force_allocation` is dissatisfying here -- I am not sure how often this will be called, but this might regress using the non-allocation 'optimized' representation in many cases.

You can check `write_immediate_no_validate` for how to write to a `PlaceTy` without `force_allocation`. (That might also help with the perf regression.)

Reply:
I've changed this in the updated code, let me know if it looks better